### DISCUSSION TITLE
SFTPSensor for sensing a file pattern

### DISCUSSION THREAD

**Discussion by AnudeepKonaboina** [apache/airflow#19928] [Q&A]
Hi Team,

The SFTPSensor of airflow only senses the file if the exact file name is present .If we give a pattern for example **<path>/*.csv** ,then it just keeps on poking for the file.

Example:
File at SFTP path :/path/abc.csv
Airflow DAG       :
```
dag = DAG(
    dag_id='test_dag',
    max_active_runs=1,
    default_args=default_args,
    catchup=False,
    schedule_interval=None
)

task_for_sensing_file = SFTPSensor(
    task_id='sense_file'
    , path="path/*.csv"
    , sftp_conn_id='sftp_conn'
    , poke_interval=30
    , dag=dag
)
task_for_sensing_file 
```

As given above i want the abc,csv file to be sensed with the pattern *.csv .Any way of dealing with this?

**Comment by potiuk** [comment]
Not now. You'd have to implement your custom sensor for that and define it's logic. Or maybe extend existint one. The sensor is really simple and you could easily write one that lists directory and checks if a path is there (and possibly contribute it back to Airflow):

https://airflow.apache.org/docs/apache-airflow-providers-sftp/stable/_modules/airflow/providers/sftp/sensors/sftp.html#SFTPSensor

**Reply by AnudeepKonaboina** [reply]
Hello @potiuk ,
Here's the pull request!!Thank you

https://github.com/apache/airflow/pull/20005
